embedded/store/immustore.go

```diff
@@ -1942,17 +2121,47 @@ func (s *ImmuStore) ReplicateTx(exportedTx []byte, waitForIndexing bool) (*TxHea
 		return nil, ErrIllegalArguments
 	}
 
-	return s.commit(txSpec, hdr, waitForIndexing)
+	txHdr, err := s.precommit(txSpec, hdr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.durablePrecommitWHub.WaitFor(txHdr.ID, nil)
```

Comment:
Potential performance issue - we durably store each precomitted TX separately.

Reply:
actually no, the precommitted transactions are subject to grouped sync as before

Reply:
yes, but if you're a replica there's only a single replicator that writes transactions to the database - and if I understand correctly, it is doing it sequentially.